### PR TITLE
Support pass through mode

### PR DESCRIPTION
**Pull request by zhangxuhong**
If `pass-through` is enabled, then simply dump data to hdfs. This is useful for just change format, shuffle data, change num of part files, etc.

### CODE REVIEW

File: avro2tf/src/main/scala/com/linkedin/avro2tf/parsers/Avro2TFJobParamsParser.scala [hunk 1]
```diff
@@ -49,7 +49,8 @@ case class Avro2TFParams(
   partitionFieldName: String,
   termOnlyFeatureList: Boolean,
   discardUnknownEntries: Boolean,
-  enableFilterZero: Boolean
+  enableFilterZero: Boolean,
+  passThrough:Boolean
```
Comment (annyan09023) [inline avro2tf/src/main/scala/com/linkedin/avro2tf/parsers/Avro2TFJobParamsParser.scala]: Do you consider to have another job to do pass through, for example, Avro2TFDataWriter instead of having passThrough option

File: avro2tf/src/main/scala/com/linkedin/avro2tf/constants/Avro2TFJobParamNames.scala [hunk 2]
```diff
@@ -26,4 +26,5 @@ object Avro2TFJobParamNames {
   final val ENABLE_TERM_ONLY_FEATURE_LIST = "enable-term-only-feature-list"
   final val DISCARD_UNKNOWN_ENTRIES = "discard-unknown-entries"
   final val ENABLE_FILTER_ZERO = "enable-filter-zero"
+  final val PASS_THROUGH = "pass-through"
```
Comment (haitong) [inline avro2tf/src/main/scala/com/linkedin/avro2tf/constants/Avro2TFJobParamNames.scala]: Please add document for this parameter, including what are the possible values, and what each value do.

We should add document for all parameters but that can go to a separate pull request

File: avro2tf/src/main/scala/com/linkedin/avro2tf/parsers/Avro2TFJobParamsParser.scala (hunk 1 above)
Comment (zhangxuhong) [reply avro2tf/src/main/scala/com/linkedin/avro2tf/parsers/Avro2TFJobParamsParser.scala]: Yes, A separate job is fine. However, to easily utilize the new job, we also need to make changes in our gradle plugin, which i think is not necessary for this simple feature support.

File: avro2tf/src/main/scala/com/linkedin/avro2tf/constants/Avro2TFJobParamNames.scala (hunk 2 above)
Comment (zhangxuhong) [reply avro2tf/src/main/scala/com/linkedin/avro2tf/constants/Avro2TFJobParamNames.scala]: Each of the params have explanations in their corresponding parse class.

File: avro2tf/src/main/scala/com/linkedin/avro2tf/parsers/Avro2TFJobParamsParser.scala (hunk 1 above)
Comment (annyan09023) [reply avro2tf/src/main/scala/com/linkedin/avro2tf/parsers/Avro2TFJobParamsParser.scala]: changing template is fine. My concern is if passThrough is true, you don't need any of other config params at all (I think that's why you make json input from require to optional). But on the other hand, it looses check. For example, if passthrough is false, you do need to provide json config. Let's not create a monolithic job with multiple branches and only use config to control it. What do you think?

Comment (zhangxuhong) [reply avro2tf/src/main/scala/com/linkedin/avro2tf/parsers/Avro2TFJobParamsParser.scala]: Many parameters are shared, i don't adding such a simple logic makes the job complicated. I do add check if passthrough mode is not enabled. Also changing the template is not that easy. It would take a least two full days to add a new template. For such simple feature support, adding a new template is really not necessary.

Comment (annyan09023) [reply avro2tf/src/main/scala/com/linkedin/avro2tf/parsers/Avro2TFJobParamsParser.scala]: I see. In that sense (it shares a lot of other params), I'm fine putting them in a job.

File: avro2tf/src/main/scala/com/linkedin/avro2tf/constants/Avro2TFJobParamNames.scala (hunk 2 above)
Comment (annyan09023) [inline avro2tf/src/main/scala/com/linkedin/avro2tf/constants/Avro2TFJobParamNames.scala]: maybe better to call it "pass-through-only" to differentiate pass-through-columns in the future if we want to have

File: avro2tf/src/main/scala/com/linkedin/avro2tf/helpers/Avro2TFJobHelper.scala [hunk 3]
```diff
@@ -213,10 +213,13 @@ object Avro2TFJobHelper {
           // Construct a new array of String array column for String array data type to be used with SequenceExample
           newConvertedColumns.append(wrapStringArray(dataFrame(columnName)).name(columnName))
           convertedColumnNames.add(columnName)
+        } else if (!CommonUtils.isValidTFRecordType(dataFrameSchema(columnName).dataType)) {
```
Comment (annyan09023) [inline avro2tf/src/main/scala/com/linkedin/avro2tf/helpers/Avro2TFJobHelper.scala:216]: why?

File: avro2tf/src/main/scala/com/linkedin/avro2tf/constants/Avro2TFJobParamNames.scala (hunk 2 above)
Comment (zhangxuhong) [reply avro2tf/src/main/scala/com/linkedin/avro2tf/constants/Avro2TFJobParamNames.scala]: sure

File: avro2tf/src/main/scala/com/linkedin/avro2tf/helpers/Avro2TFJobHelper.scala (hunk 3 above)
Comment (zhangxuhong) [reply avro2tf/src/main/scala/com/linkedin/avro2tf/helpers/Avro2TFJobHelper.scala:216]: adding a safety check, TFRecord only supports primitive and array and arrayofarray types
